### PR TITLE
Complete Marketplace Design

### PR DESCRIPTION
**Pull request by dherault**
## Summary

Resolves ENG-118

Finalizes the layout for the "Marketplace" and "Installed" scenes.

image
## Checklist

- [x] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [x] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: www/src/components/marketplace/MarketplaceRepositories.js [hunk 1]
```diff
@@ -71,6 +78,14 @@ function MarketplaceRepositories({ installed }) {
     })
     .filter(repository => installed ? repository.installation : true)
 
+  const fuse = new Fuse(sortedRepositories, searchOptions)
+
+  const resultRepositories = search ? fuse.search(search).map(({ item }) => item) : sortedRepositories
```
[the remark this answers is not in the document]
Comment (dherault) [reply www/src/components/marketplace/MarketplaceRepositories.js:79]: Yes, this hack works well for now.
In the future we might want to use the open-source version of Algolia 🇫🇷 !

[the remark this answers is not in the document]
Comment (michaeljguarino) [reply www/src/components/marketplace/MarketplaceRepositories.js:79]: no we should use the api's own search features.

File: www/src/components/ModalHeader.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ export function ModalHeader({ text, setOpen }) {
           justify="center"
           width="2rem"
           height="2rem"
-          hoverIndicator="background-light"
+          hoverIndicator="fill-one"
```
Comment (michaeljguarino) [inline www/src/components/ModalHeader.js:24]: the name fill-one seems a little unclear to me, is there a way to make it a little more apparent what that color is acting as?

File: www/src/components/marketplace/Marketplace.js [hunk 3]
```diff
@@ -53,32 +53,30 @@ function Marketplace({ installed }) {
...
+          transform={areFiltersOpen ? 'translateX(0)' : 'translateX(100%)'}
+          opacity={areFiltersOpen ? 1 : 0}
           flexShrink={0}
           position="sticky"
           top={0}
-          left={0}
-          width={areFiltersOpen ? 256 - 32 : 0}
-          height={`calc(100% - ${32}px)`}
+          right={0}
+          width={256 - 32}
```
Comment (michaeljguarino) [inline www/src/components/marketplace/Marketplace.js:73]: is there a reason why we don't just make thse constants?

File: www/src/components/marketplace/Marketplace.js [hunk 4]
```diff
@@ -53,32 +53,30 @@ function Marketplace({ installed }) {
...
+          height={`calc(100% - ${64 + 8}px)`}
```
Comment (michaeljguarino) [inline www/src/components/marketplace/Marketplace.js:74]: same shouldn't this just be a const?

File: www/src/components/marketplace/MarketplaceSidebar.js [hunk 5]
```diff
@@ -59,96 +52,127 @@ function MarketplaceSidebar(props) {
...
+    )
   }
 
-  const sortedCategories = categoriesData.categories.slice().sort((a, b) => a.category.localeCompare(b.category))
-  const sortedPublishers = ['Plural']
-  const sortedTags = tags.slice().sort((a, b) => a.tag.localeCompare(b.tag)).filter((x, i) => i < nDisplayedTags)
+  function renderTags() {
+    const sortedTags = tags.slice().sort((a, b) => a.name.localeCompare(b.name))
+    const fuse = new Fuse(sortedTags, searchOptions)
+    const resultTags = search ? fuse.search(search).map(({ item }) => item) : sortedTags.filter((x, i) => i < nDisplayedTags)
```
Comment (michaeljguarino) [inline www/src/components/marketplace/MarketplaceSidebar.js:121]: we're definitely not going to be able to load all tags in one fetch in relatively short order, so this should really be moved to utilizing the gql apis properly

File: www/src/components/ModalHeader.js (hunk 2 above)
Comment (dherault) [reply www/src/components/ModalHeader.js:24]: Ask Nick, that's his responsibility

File: www/src/components/marketplace/Marketplace.js (hunk 3 above)
Comment (dherault) [reply www/src/components/marketplace/Marketplace.js:73]: I will

File: www/src/components/marketplace/MarketplaceSidebar.js (hunk 5 above)
Comment (dherault) [reply www/src/components/marketplace/MarketplaceSidebar.js:121]: I'm waiting on some green go from the backend team then.

Comment (michaeljguarino) [reply www/src/components/marketplace/MarketplaceSidebar.js:121]: what do you need api wise?  I'm pretty sure we do have the capability to filter tags

File: www/src/components/ModalHeader.js (hunk 2 above)
Comment (michaeljguarino) [reply www/src/components/ModalHeader.js:24]: We can bring it up with nick too, but for codebase cleanliness, I do think we should have a mapping of semantic names that correspond to his more systematic ones.

File: www/src/components/marketplace/MarketplaceSidebar.js (hunk 5 above)
Comment (michaeljguarino) [reply www/src/components/marketplace/MarketplaceSidebar.js:121]: yep, you can use the `q` argument for the tags query to filter by a search string

File: www/src/components/marketplace/MarketplaceRepositories.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaeljguarino) [reply www/src/components/marketplace/MarketplaceRepositories.js:79]: You should now have the `categories`, `publishers` and `tags` fields to do multi-value filtering for all of these via the api.  We should definitely strongly prefer that to in-browser filtering

File: www/src/components/marketplace/MarketplaceRepositories.js [hunk 6]
```diff
@@ -61,16 +64,33 @@ function MarketplaceRepositories({ installed }) {
 
   const sortedRepositories = repositories.slice()
     .sort((a, b) => a.name.localeCompare(b.name))
-    .filter(repository => categories.length ? categories.includes(repository.category) : true)
+    .filter(repository => categories.length ? categories.includes(repository.category.toLowerCase()) : true)
```
Comment (michaeljguarino) [inline www/src/components/marketplace/MarketplaceRepositories.js:67]: sorting should also be delegated to the api

Comment (dherault) [reply www/src/components/marketplace/MarketplaceRepositories.js:67]: I've added a ticket for you to do it.

**Comment by dogmar** [comment]
Scrollbar should appear on right edge of browser window. Otherwise, looks great.
Screen Shot 2022-06-07 at 11 05 04 AM

**Comment by dherault** [comment]
Will work on it.

**Comment by dogmar** [comment]
Marketplace scrollbar is good. I should have caught this earlier, but Filters sidebar is overflowing horizontally and showing a horizontal scrollbar.

Screen Shot 2022-06-07 at 1 12 42 PM

**Review by michaeljguarino — APPROVED** [review]
This looks good, make sure I have a ticket to fix up the api usage and let's press on.

**Pull request merged**